src/api/java/com/direwolf20/buildinggadgets/api/materials/UniqueItem.java

```diff
@@ -43,6 +44,10 @@ public Item getItem() {
         return item;
     }
 
+    public ItemStack toItemStack() {
+        return new ItemStack(item, 1, tagCompound);
```

Comment:
Notice that this is not what the tagCompound comes from... The last Parameter to ItemStacks is the Capability NBT...

I've added an `createItemStack` Method to help with that, as well as taking the difference between cap nbt and share tag into account.